components/org.wso2.carbon.healthcheck.api.core/src/main/java/org/wso2/carbon/healthcheck/api/core/internal/HealthMonitorServiceComponent.java

```diff
@@ -51,6 +52,8 @@ protected void activate(ComponentContext ctxt) {
                     new ServerStartupChecker(), null);
             ctxt.getBundleContext().registerService(HealthChecker.class.getName(),
                     new DataSourceHealthChecker(), null);
+            ctxt.getBundleContext().registerService(CarbonMemoryAgentService.class.getName(),
```

Comment:
I don't think we need to register this as an OSGI service as this only has private methods. Why do we need this service ? The service get instance method itself registers this as a scheduled task. 

If u return this as a service, I think there should be a different method to start this as a task.

Reply:
Actually I was under the impression that the core functionalities will be given as a service and it will be invoked by a health checker impl class. So the health checker impl class has required input configurations and so on.. This way we don't need to add additional cofig elements as well I think

Reply:
I have made the getMemoryUtilization method public so that it can be called from elsewhere such as health check impl. The impl class here is responsible for building the health check api response. Since this memory data does not getting revealed through the API, I have just kept the method public.

Reply:
Then we should give the facility to configure timer task. Otherwise it might be an unnecessary thing.